aiohttp/helpers.py

```diff
@@ -723,10 +734,16 @@ def __exit__(
         exc_val: Optional[BaseException],
         exc_tb: Optional[TracebackType],
     ) -> Optional[bool]:
+        enter_task: Optional[asyncio.Task[Any]] = None
         if self._tasks:
-            self._tasks.pop()  # type: ignore[unused-awaitable]
+            enter_task = self._tasks.pop()
 
         if exc_type is asyncio.CancelledError and self._cancelled:
+            assert enter_task is not None
+            # The timeout was hit, and the task was cancelled
+            # so we need to uncancel the last task that entered the context manager
+            # since the cancellation should not leak out of the context manager
+            _uncancel_task(enter_task)
```

Comment:
Isn't it easier to do version check here, and not end up with a pointless function after pyupgrade removes the duplicate?

Reply:
I copied it from async timeout. Happy to inline it instead